aptos-move/move-examples/fungible_asset/sources/managed_fungible_asset.move

```diff
@@ -0,0 +1,216 @@
+module fungible_asset::managed_fungible_asset {
+    use std::option;
+    use aptos_framework::object::{Object, object_address, ExtendRef, is_owner};
+    use fungible_asset::fungible_source::{MintCap, FreezeCap, BurnCap, unfreeze_with_cap, burn_with_cap, mint_with_cap, init_fungible_source, freeze_with_cap};
+    use std::option::Option;
+    use aptos_framework::object;
+    use std::error;
+    use std::signer::address_of;
+    use fungible_asset::fungible_source;
+    #[test_only]
+    use fungible_asset::fungible_asset::{create_test_token, balance, is_frozen};
+    #[test_only]
+    use std::signer;
+    #[test_only]
+    use aptos_framework::object::generate_extend_ref;
+
+    /// Mint capability exists or does not exist.
+    const EMINT_CAP: u64 = 1;
+    /// Freeze capability exists does not exist.
+    const EFREEZE_CAP: u64 = 2;
+    /// Burn capability exists or does not exist.
+    const EBURN_CAP: u64 = 3;
+    /// Not the owner.
+    const ENOT_OWNER: u64 = 4;
+    /// Caps existence errors.
+    const EMANAGED_FUNGIBLE_ASSET_CAPS: u64 = 5;
+
+    struct Caps has key {
+        mint: Option<MintCap>,
+        freeze: Option<FreezeCap>,
+        burn: Option<BurnCap>,
+    }
+
+    public fun initialize_managing_capabilities(
+        extend_ref: &ExtendRef,
```

Comment:
I don't think we should support creating an asset after construction, prefer this to be ConstructorRef